product_variant_configurator_manual_creation/wizards/wizard_product_variant_configurator_manual_creation.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2022 ForgeFlow S.L. <https://forgeflow.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from itertools import product
+
+from odoo import api, fields, models
+
+
+class WizardProductVariantConfiguratorManualCreation(models.TransientModel):
+    _name = "wizard.product.variant.configurator.manual.creation"
+    _description = "Manual Creation Variant Configurator"
+
+    product_tmpl_id = fields.Many2one(
+        comodel_name="product.template", string="Template", readonly=True
+    )
+    variants_to_create = fields.Integer(compute="_compute_variants_to_create")
+    line_ids = fields.One2many(
+        comodel_name="wizard.product.variant.configurator.manual.creation.line",
+        inverse_name="wizard_id",
+        string="Lines",
+        required=False,
+    )
+
+    @api.depends(
+        "product_tmpl_id",
+        "line_ids.selected_value_ids",
+    )
+    def _compute_variants_to_create(self):
+        for rec in self:
+            combinations = rec._get_combinations()
+            rec.variants_to_create = len(combinations)
+
+    def _get_combinations(self):
+        self.ensure_one()
+        selected_items = [
+            line.selected_value_ids.ids
+            for line in self.line_ids
+            if line.selected_value_ids
+        ]
+        return list(selected_items and product(*selected_items))
+
+    @api.model
+    def default_get(self, fields_list):
+        values = super().default_get(fields_list)
+        values["product_tmpl_id"] = self.env.context.get("active_id")
+        return values
```

Comment:
<img width="1916" height="860" alt="Image" src="https://github.com/user-attachments/assets/62fd121a-5d6c-4ab5-9a59-fcbda4d5ef71" />

The button is in variants (which is useful), but in default_get you're trying to open it only from the product_template. You should check active_model.
```suggestion
    @api.model
    def default_get(self, fields_list):
        values = super().default_get(fields_list)
        product_tmpl_id = False
        if self._context.get("active_model") == "product.template":
            product_tmpl_id = self.env.context.get("active_id")
        elif self._context.get("active_model") == "product.product":
            product_tmpl_id = self.env["product.product"].browse(
                self.env.context.get("active_id")
            ).product_tmpl_id.id
        values["product_tmpl_id"] = product_tmpl_id
        return values
```

Reply:
You're right. This bug was already present in v15, hence is independent of the migration. For more clarity I've fixed it in a new commit.
Edit: I must check why tests fail, I'll do it asap.

Reply:
Hi, I've fixed the tests. The active_model must now be specified in tests. Moreover I've added a test to simulate the opening of the wizard from the product.product.